docs/cookbook/recipe_icheck.rst

```diff
@@ -43,3 +43,32 @@ set data attribute ``data-sonata-icheck = "false"`` to this form element::
 .. note::
 
     You have to use false as string! ``"false"``!
+
+When using ``Symfony\Component\Form\Extension\Core\Type\ChoiceType`` (or other types that inherit from it, such as )
+``Sonata\AdminBundle\Form\Type\ChoiceFieldMaskType``)
+with the ``expanded``: ``true`` option, it is necessary to set the ``data-sonata-icheck`` attribute
+both on the form type and on the choice items::
+
+    use Sonata\AdminBundle\Form\FormMapper;
+    use Sonata\AdminBundle\Form\Type\ModelType;
+    use Sonata\AdminBundle\Form\Type\ChoiceFieldMaskType;
+
+    protected function configureFormFields(FormMapper $form): void
+    {
+        $form
+            ->add('category', ChoiceFieldMaskType::class, [
+                'expanded' => true,
+                'attr' => [
+                    'data-sonata-icheck' => 'false'
+                ],
+                'placeholder_attr' => [ // the placeholder if any needs also the data-sonata-icheck attr
+                    'data-sonata-icheck' => 'false'
+                ],
+                'choice_attr' => [
+                    'val1' => ['data-sonata-icheck' => 'false'],
+                    'val2' => ['data-sonata-icheck' => 'false'],
+                    // ...
+                ],
```

Comment:
It would be better to avoid all this.
Cant we access the `attr` on the field in the twig instead of every choice ?

Also what if we always listen on both `ifChecked` and `change` ? Could it avoid the issue ?

Reply:
Actually listening on both events seems a great idea. Will try it tomorrow